pandas/tests/test_window.py

```diff
@@ -2713,3 +2713,129 @@ def test_rolling_min_max_numeric_types(self):
             result = (DataFrame(np.arange(20, dtype=data_type))
                       .rolling(window=5).min())
             self.assertEqual(result.dtypes[0], np.dtype("f8"))
+
+
+class Grouping(tm.TestCase):
+
+    def setUp(self):
+        self.series = Series(np.arange(10))
+        self.frame = DataFrame({'A': [1] * 20 + [2] * 12 + [3] * 8,
+                                'B': np.arange(40)})
+
+    def test_mutated(self):
+
+        def f():
+            self.frame.groupby('A', foo=1)
+        self.assertRaises(TypeError, f)
+
+        g = self.frame.groupby('A')
+        self.assertFalse(g.mutated)
+        g = self.frame.groupby('A', mutated=True)
+        self.assertTrue(g.mutated)
+
+    def test_getitem(self):
+        g = self.frame.groupby('A')
+        g_mutated = self.frame.groupby('A', mutated=True)
+
+        expected = g_mutated.B.apply(lambda x: x.rolling(2).mean())
```

Comment:
The fundamental thing is its very tricky to infer whether to create mutli-indexes with a user passed function. So I think that we 'guess' by seeing if it is indexed the same or not (e.g. mutated). Can't really fault the logic. With the new ``.rolling/expanding/resample`` we KNOW that we need to produce a MI.